plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt

```diff
@@ -36,7 +37,9 @@ import software.aws.toolkits.telemetry.FeatureId
 import java.awt.event.ComponentAdapter
 import java.awt.event.ComponentEvent
 
-class AmazonQToolWindowFactory : ToolWindowFactory, DumbAware {
+class AmazonQToolWindowFactory(
+    private val disposable: Disposable,
+) : ToolWindowFactory, DumbAware {
```

Comment:
i don't believe the platform supports injecting the disposable like this. `ToolWindowFactory` is supposed to be an application-global stateless singleton, so i think you would want to fetch the disposable instance from the tool window instead.